src/purple/blist.rs

```diff
@@ -0,0 +1,40 @@
+use super::ffi::AsPtr;
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::ptr::NonNull;
+
+pub struct BlistNode(NonNull<purple_sys::PurpleBlistNode>);
+
+impl BlistNode {
+    pub unsafe fn from_ptr(ptr: *mut purple_sys::PurpleBlistNode) -> Option<Self> {
+        NonNull::new(ptr).map(Self)
+    }
+
+    pub fn get_string(&mut self, c_key: &CString) -> Option<String> {
+        let c_str =
+            unsafe { purple_sys::purple_blist_node_get_string(self.0.as_ptr(), c_key.as_ptr()) };
+        if c_str.is_null() {
```

Comment:
```
            NonNull::new(c_str)
            .map(|p| {
                CStr::from_ptr(p.as_ptr() as *const c_char)
                    .to_str()
                    .unwrap()
            })
```

? (This would return an `Option<&str>`)

Reply:
I can't seem to create the NonNull with the c_value.

Am I doing something wrong?

```rust
    pub fn get_string(&mut self, key: &str) -> Option<&str> {
        let c_key = CString::new(key).unwrap();
        let mut c_value =
            unsafe { purple_sys::purple_blist_node_get_string(self.0.as_ptr(), c_key.as_ptr()) };

        NonNull::new(c_value).map(|p| {
            CStr::from_ptr(p.as_ptr() as *const c_char)
                .to_str()
                .unwrap()
        })
    }
```

```
   = note: expected raw pointer `*mut _`
              found raw pointer `*const i8`
```

Reply:
Just add `NonNull::new(c_value as *mut c_char)` I believe.

Reply:
Done!